packages/authorization-platformstate-writer/src/consumerServiceV1.ts

```diff
@@ -1,20 +1,512 @@
+  convertEntriesToClientKidInTokenGenerationStates,
+  deleteClientEntryFromPlatformStates,
+  deleteClientEntryFromTokenGenerationStatesTable,
+  deleteEntriesFromTokenStatesByClient,
+  deleteEntriesFromTokenStatesByKid,
+  deleteEntriesWithClientAndPurposeFromTokenGenerationStatesTable,
+  extractAgreementIdFromAgreementPK,
+  extractKidFromTokenEntryPK,
+  readClientEntriesInTokenGenerationStates,
+  readClientEntry,
+  retrievePlatformStatesByPurpose,
+  setClientPurposeIdsInPlatformStatesEntry,
+  upsertPlatformClientEntry,
+  upsertTokenClientKidEntry,
+  upsertTokenStateClientPurposeEntry,
+} from "./utils.js";
 
 export async function handleMessageV1(
   message: AuthorizationEventEnvelopeV1,
-  _dynamoDBClient: DynamoDBClient
+  dynamoDBClient: DynamoDBClient
 ): Promise<void> {
   await match(message)
+    .with({ type: "ClientAdded" }, async (msg) => {
+      const client = parseClient(msg.data.client, msg.type);
+
+      const pk = makePlatformStatesClientPK(client.id);
+      const clientEntry = await readClientEntry(pk, dynamoDBClient);
+
+      if (clientEntry && clientEntry.version > msg.version) {
+        return Promise.resolve();
+      } else {
+        const updatedClientEntry: PlatformStatesClientEntry = {
+          PK: pk,
+          version: msg.version,
+          state: itemState.active,
+          clientKind: clientKindToTokenGenerationStatesClientKind(client.kind),
+          clientConsumerId: client.consumerId,
+          updatedAt: new Date().toISOString(),
+          clientPurposesIds: client.purposes,
+        };
+        await upsertPlatformClientEntry(updatedClientEntry, dynamoDBClient);
+      }
+    })
+    // eslint-disable-next-line sonarjs/cognitive-complexity
+    .with({ type: "KeysAdded" }, async (msg) => {
+      const keyV1 = msg.data.keys[0].value;
+
+      const clientId = unsafeBrandId<ClientId>(msg.data.clientId);
+      const key = parseKey(keyV1, msg.type);
+      const kid = key.kid;
+
+      const pem = key.encodedPem;
+      if (!pem) {
+        throw missingKafkaMessageDataError("key", msg.type);
+      }
+
+      const pk = makePlatformStatesClientPK(clientId);
+      const clientEntry = await readClientEntry(pk, dynamoDBClient);
+
+      if (!clientEntry || clientEntry.version > msg.version) {
+        return Promise.resolve();
+      } else {
+        const clientPurposesIds = clientEntry?.clientPurposesIds || [];
+
+        const platformClientEntry: PlatformStatesClientEntry = {
+          PK: pk,
+          state: itemState.active,
+          clientKind: clientEntry.clientKind,
+          clientConsumerId: clientEntry.clientConsumerId,
+          clientPurposesIds,
+          version: msg.version,
+          updatedAt: new Date().toISOString(),
+        };
+        await upsertPlatformClientEntry(platformClientEntry, dynamoDBClient);
+
+        if (platformClientEntry.clientPurposesIds.length > 0) {
+          const map: Map<PurposeId, TokenGenerationStatesClientPurposeEntry> =
```

Comment:
same considerations [here](https://github.com/pagopa/interop-be-monorepo/pull/1062#discussion_r1819095340)

Reply:
Updated